realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -2976,4 +2977,187 @@ public void clear_all() {
         assertEquals(0, realm.where(Cat.class).count());
         assertTrue(realm.isEmpty());
     }
+
+    // Test if waitForChange gets waked up by 1) empty transaction 2) transaction with some data changes.
+    @Test
+    public void waitForChange() throws InterruptedException {
```

Comment:
I don't think we should duplicate these tests across RealmTests / DynamicRealmTests, they become impossible to maintain. Either just delete one of them or better yet, convert it to a parameterized test (we should probably do that anyway for a lot of methods, so this would be a nice chance to begin that process).